BLSQ/iaso
# WC2-61 WC2-70 WC2-64:  NFC

Storage entry in menu, custom api, ability to black list a storage with a reason.

## Self proof reading checklist

## Changes

 - list storages on a page, filter, sort and pagination are working [WC2-61](https://bluesquare.atlassian.net/browse/WC2-61)
 - storage details page with list of logs [WC2-70](https://bluesquare.atlassian.net/browse/WC2-70)
 - allow to blacklist a storage for a specific reason (front & back) [WC2-64](https://bluesquare.atlassian.net/browse/WC2-64)


## How to test

Create storages and logs via the admin
Make sure you are super admin or that you have 'iaso_storages' permissions and visit the storage page.

## Print screen / video

<img width="775" alt="Screenshot 2022-10-20 at 11 09 29" src="https://user-images.githubusercontent.com/12494624/196907004-0da8ddd5-4b7a-44e3-a68d-abb639d3c096.png">
<img width="1673" alt="Screenshot 2022-10-20 at 11 09 23" src="https://user-images.githubusercontent.com/12494624/196907012-48087879-fdda-4108-987c-f77b69540890.png">
<img width="1672" alt="Screenshot 2022-10-20 at 11 09 16" src="https://user-images.githubusercontent.com/12494624/196907017-dc561f90-38d8-4aec-a13f-b0cefb8eaf7f.png">

## Notes

Still to do:
- paginate, order and filter logs on detail page [WC2-92](https://bluesquare.atlassian.net/browse/WC2-92) 
- exports in CSV/XLS for list and details logs [WC2-93](https://bluesquare.atlassian.net/browse/WC2-93) and [WC2-94](https://bluesquare.atlassian.net/browse/WC2-94)
- search on list page should search in storage id, entity attributes and org unit [WC2-95](https://bluesquare.atlassian.net/browse/WC2-95)
